misk-crypto/src/main/kotlin/misk/crypto/CryptoModule.kt

```diff
@@ -72,16 +75,24 @@ class CryptoModule(
     }
   }
 
-  private class AeadProvider(val keyUri: String, val key: Key) : Provider<Aead> {
+  /**
+   * We do not support creating bare AEAD keys, only via envelope encryption.
+   */
+  private class AeadEnvelopeProvider(val keyUri: String, val key: Key) : Provider<Aead> {
     @Inject lateinit var keyManager: AeadKeyManager
     @Inject lateinit var kmsClient: KmsClient
 
     override fun get(): Aead {
       val keysetHandle = readKey(key.encrypted_key, kmsClient.getAead(keyUri))
-      return AeadFactory.getPrimitive(keysetHandle)
-          .also { keyManager[key.key_name] = it }
+      val kek = AeadFactory.getPrimitive(keysetHandle)
+      val envelopeKey = KmsEnvelopeAead(DEK_TEMPLATE, kek)
+
+      return envelopeKey.also { keyManager[key.key_name] = it }
     }
 
+    companion object {
+      val DEK_TEMPLATE : KeyTemplate = AeadKeyTemplates.AES128_GCM
```

Comment:
You are not using this outside this file. Do you mean to use it in tests?

Reply:
Just wanted to keep the DEK template visually separate from its use, if we want to adjust it later. Don't mind inlining it however.